packages/n4s/src/rules/__tests__/isArrayOf.test.js

```diff
@@ -0,0 +1,40 @@
+import { isArrayOf } from 'isArrayOf';
+import enforse from 'enforce'
+
+describe('Tests isArrayOf rule', () => {
+
+  it('Should return true if all elements are ture for one or more rules', () => {
+    expect(isArrayOf([3,4,5,'six',7],
+     enforse.greaterThan(2), 
+     enforse.isString())
+     ).toBe(true);
+  });
+
+
+  it('Should return false if one element or more fails all rules', () => {
+    expect(isArrayOf([3,4,5,['s','i','x'],7], 
+    enforse.greaterThan(2), 
+    enforse.isString())
+    ).toBe(false);
+  });
+
+
+  describe('Tests for recursive call', ()=>{
+
+    it('Should return true if all elements are ture for one or more rules', () => {
+      expect(isArrayOf([3,4,5,['s','i','x'],7],
+      enforse.greaterThan(2), 
+      enforse.isArrayOf(enforse.isString()))
+      ).toBe(true);
+    });
+  
+    it('Should return false if one element or more fails all rules', () => {
+      expect(
+        isArrayOf([3,4,5,['s','i','x'],7],
+        enforse.greaterThan(2),
+        enforse.isArrayOf(enforse.isNumber()))
+        ).toBe(false);
+    });
+  });
+
```

Comment:
I would also add one integration test that makes sure everything works together with enforce

Here's an example, should probably be more thorough.

```js
describe('as part of enforce', () => {
  it('should return silently when valid', () => {
    enforce([1,2,'3']).isArrayOf(enforce.isNumber())
    enforce([1,2,'3']).isArrayOf(enforce.isString())
    enforce([1,2,'3']).isArrayOf(enforce.isNumeric(), enforce.lessThan(5).greaterThan(0))
  });

  it('should throw an exception when invalid', () => {
    expect(() => enforce([1,2,'3']).isArrayOf(enforce.isNull())).toThrow()
    enforce([1,2,'3']).isArrayOf(enforce.isNumber(), enforce.greaterThan(5))
  });
})
```

Reply:
Thank you for the replay,
I'm working on it, and I have a question
in the `'as part of enforce' -> 'should return silently when valid'` test is 
```
enforce([1,2,'3']).isArrayOf(enforce.isNumber())
enforce([1,2,'3']).isArrayOf(enforce.isString())
```
need to return silently? '3' isn't  a number and '1','2' isn't a string.
did you meant to write
`enforce([1,2,'3']).isArrayOf(enforce.isNumber(), enforce.isString())`?

Reply:
Ahh, right. You are correct. I just wrote it as an example. Should be more similar to what you wrote.